src/libraries/System.Data.Common/src/System/Xml/XmlDataDocument.cs

```diff
@@ -2553,8 +2551,19 @@ public XmlDataDocument(DataSet dataset) : base(new XmlDataImplementation())
 
         internal XmlDataDocument(XmlImplementation imp) : base(imp)
         {
+            _dataSet = null!;
+            _pointers = null!;
+            _columnChangeList = null!;
+            _mapper = null!;
+            _foliationLock = null!;
+            _attrXml = null!;
         }
 
+        [MemberNotNull(nameof(_pointers))]
+        [MemberNotNull(nameof(_columnChangeList))]
+        [MemberNotNull(nameof(_mapper))]
+        [MemberNotNull(nameof(_foliationLock))]
+        [MemberNotNull(nameof(_attrXml))]
```

Comment:
You can also use the ctor that takes a params array (though you'll likely also need to suppress a CLS compliance warning).

Reply:
(Random side question: why do we get CLS compliance warnings on attributes of non-public members?)

Reply:
AFAIK CLS compliance is about consistent attribution and doesn't take visibility into consideration. If you say "the entire assembly/type is CLS compliant", then the compiler will force you to mark the ones that aren't, no matter their visibility. Presumably, that was done to make the rules simpler.

Reply:
> You can also use the ctor that takes a params array (though you'll likely also need to suppress a CLS compliance warning).

I noticed the CLS compliance warning and looked around what other parts did. The other locations in S.D.C seemed use multiple applications, so that's what I did. Do you want me to change them all?

Reply:
> AFAIK CLS compliance is about consistent attribution and doesn't take visibility into consideration.

That isn't true for method return types:

![image](https://user-images.githubusercontent.com/8291187/91750860-6bcf0e00-eb89-11ea-9b3e-0443d9c11b87.png)

Reply:
> Do you want me to change them all?

You don't need to. I was just calling out the alternative.

Reply:
> > AFAIK CLS compliance is about consistent attribution and doesn't take visibility into consideration.
> 
> That isn't true for method return types:

OK, now I'm confused :-)

Reply:
Me too. ;)